Add ARA records ansible

OgarOgarovic opened the pull request:
Adds the option to deploy [ARA records ansible](https://ara.recordsansible.org/) recording all ansible output on the host and also inside VMs, providing a HTTP UI on the host.

OgarOgarovic committed:
Working local deployment

Signed-off-by: Ivan Vnučko <ivan@vnucko.com>

OgarOgarovic committed:
Working for control and deployment

Signed-off-by: Ivan Vnučko <ivan@vnucko.com>

MatusJenca2 commented on README.md:
```diff
@@ -56,6 +57,19 @@ pip install lxml
+#### Deploying ARA records ansible server
+
+To deploy ARA server and start recording host ansible plays with ensure flag
+`ara_enable` in `globals.yml` is set to true and run:
```
I don't think `globals.yml` is where ARA should be configured.
`globals.yml` is for kolla-ansible configuration.

OgarOgarovic replied:
Wrong info in README, should be user_config, I'll fix it.

MatusJenca2 commented on README.md:
````diff
@@ -56,6 +57,19 @@ pip install lxml
+
+```bash
+./builder ara user/local-aio.yml
+```
````
```
    The conditional check 'ara_enable | bool' failed. The error was: error while evaluating conditional (ara_enable | bool): 'ara_enable' is undefined
```
Please add default value to `group_vars: all`

OgarOgarovic replied:
Added...

MatusJenca2 commented on spawn.yml:
```diff
@@ -9,3 +9,45 @@
     - name: create_multinode
       when: not is_aio
+
+  tasks:
```
I dont understand why you put this here and not prepare playbook.

OgarOgarovic replied:
Valid point, moved it to prepare playbook.

MatusJenca2 commented on spawn.yml:
```diff
@@ -9,3 +9,45 @@
+  hosts: ara_deployments
+  user: kolla
+  gather_facts: false
+
```
```suggestion
    vars:
      ansible_ssh_common_args: '-o IdentitiesOnly=yes'
      
```
Or just move all this to `prepare` playbook.

OgarOgarovic replied:
Moved it to prepare playbook.

MatusJenca2 commented on roles/config_ara_recording/tasks/main.yml:
```diff
@@ -0,0 +1,76 @@
+
+- name: Get ARA callback plugins definition
+  ansible.builtin.shell: |
+    . {{ (python_bin_path, 'activate') | path_join }}
```
This works only in a virtual environment, not with natively installed python i.e. `/usr/bin/python`
```
TASK [config_ara_recording : Get ARA callback plugins definition] 
fatal: [localhost]: FAILED! => changed=true 
  cmd: |-
    . /usr/bin/activate
    python3 -m ara.setup.callback_plugins
  delta: '0:00:00.002860'
  end: '2024-05-29 13:22:11.322442'
  msg: non-zero return code
  rc: 127
  start: '2024-05-29 13:22:11.319582'
  stderr: '/bin/sh: 1: .: Can''t open /usr/bin/activate'
  stderr_lines: <omitted>
  stdout: ''
  stdout_lines: <omitted>

```

OgarOgarovic replied:
Removed the dependency on virtual environment activation.

MatusJenca2 commented on ara.yml:
```diff
@@ -0,0 +1,16 @@
+---
```
How does this work on a remote libvirt host?

OgarOgarovic replied:
Updated to properly install on local or remote host.

OgarOgarovic committed:
Updates and fixes from review

- add ara_enable flag to group_vars/all.yml
- moved ARA recording setup to prepare playbook
- fixed README
- fixed virtual environment dependecy
- fixed deploy to remote

MatusJenca2 commented on spawn.yml:
```diff
@@ -9,3 +9,45 @@
+#         ara_server_ip: "{{ ansible_facts['env']['SSH_CLIENT'].split() | first }}"
+#         config_to_edit: "~/ansible.cfg"
+#         python_bin_path: "~/kolla/bin/"
+#       when: ara_enable | bool
```
Fine, just remove the commented code and we can merge.

OgarOgarovic committed:
Remove leftover commented code
